async-nats/src/auth.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020-2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(Clone)]
+pub(crate) enum Authorization {
```

Comment:
This is authentication right?

```suggestion
pub(crate) enum Authentication {
```

Reply:
It is, but related section in NATS server config is called Authorization https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/username_password

Reply:
Nevermind, docs refer to it as authorization